ouroboros-consensus-cardano-legacy-block/ouroboros-consensus-cardano-legacy-block.cabal

```
@@ -0,0 +1,74 @@
+cabal-version:   3.0
+name:            ouroboros-consensus-cardano-legacy-block
+version:         0.1.0.0
+synopsis:
+  Legacy blocks for the instantation of the Ouroboros consensus layer used by Cardano
+
+description:
+  Legacy blocks for the instantation of the Ouroboros consensus layer used by Cardano.
+
+license:         Apache-2.0
+license-files:
+  LICENSE
+  NOTICE
+
+copyright:       2023-2023 Input Output Global Inc (IOG)
+author:          IOHK Engineering Team
+maintainer:      operations@iohk.io
+category:        Network
+build-type:      Simple
+extra-doc-files:
+  CHANGELOG.md
+  README.md
+
+source-repository head
+  type:     git
+  location: https://github.com/input-output-hk/ouroboros-consensus
+
+flag asserts
+  description: Enable assertions
+  manual:      False
+  default:     False
+
+common common-lib
+  default-language: Haskell2010
+  ghc-options:
+    -Wall -Wcompat -Wincomplete-uni-patterns
+    -Wincomplete-record-updates -Wpartial-fields -Widentities
+    -Wredundant-constraints -Wmissing-export-lists -Wunused-packages
+    -Wno-unticked-promoted-constructors
```

```suggestion
    -Wall -Wcompat -Wpartial-fields -Wmissing-export-lists -Wunused-packages
```
Following the user guide I think the others are already set by `Wall`

---

And the same can probably be done also in other packages

---

https://downloads.haskell.org/~ghc/latest/docs/users_guide/using-warnings.html?highlight=warnings#ghc-flag--Wall

---

What's included in `-Wall` can change between GHC versions, so just to be safe we could keep these explicit flags 

Compare: https://downloads.haskell.org/ghc/8.10.7/docs/html/users_guide/using-warnings.html